samcli/lib/package/utils.py

```diff
@@ -280,8 +279,12 @@ def make_zip_with_permissions(file_name, source_root, permission_mappers: List[P
                             info.external_attr = os.stat(full_path).st_mode << 16
                             for permission_mapper in permission_mappers:
                                 info = permission_mapper.apply(info)
-                            # Set current time to be the last time the zip content was modified.
-                            info.date_time = time.localtime()[0:6]
+                            # ZIP date time can be set to the last time the zip content was modified using this logic.
+                            # info.date_time = time.localtime()[0:6]
+
+                            # If the date time above is added, the caching logic that compares ZIP files sha will break.
```

Comment:
Can we be more specific on the caching logic and where it is used.

Reply:
Could we also add a unit test here?

Reply:
Updated with a new unit test